library/Zend/Http/Client.php

```diff
@@ -1192,6 +1192,13 @@ protected function _prepareHeaders()
             }
         }
 
+        // Set the request id header if the unique id environment variable
+        // is set.
+        $uniqueId = getenv('UNIQUE_ID');
+        if ($uniqueId && !isset($this->headers['x-request-id'])) {
+            $headers[] = "x-request-id: {$uniqueId}";
```

Comment:
So this method forms a $headers array in which it places a couple of headers (like this one). Eventually, it rolls the full $this->headers array into the $headers array and then returns it.